modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenDiscriminator.java

```diff
@@ -72,6 +72,11 @@ public MappedModel(String mappingName, String modelName) {
             this.modelName = modelName;
         }
 
+        @Override
+        public int compareTo(MappedModel other) {
+            return getMappingName().compareTo(other.getMappingName());
```

Comment:
Yes, please submit a PR to my branch. I saw that you threw the error at assignment mappingName which is preferable to throwing the error later.
The only null that I saw was in a test. Why would it be null?

Reply:
github is not allowing me to open a PR against your fork. I have this fix in my branch, I will ask William to merge after you. This happens for example when there is at least one vendor extension but not the discriminator extension. Also, since I don't think mappingName should be null, it would be good to catch early to make it easier to troubleshoot issues.

Reply:
This PR now works for the use case where there is at least one vendor extension but not the discriminator extension. Let's handle the null issue in your PR. I'd prefer not to have this very large PR grow in any more scope.

Reply:
@sebastien-rosset please include me in the review process of your later PR.

Reply:
So it looks like I have more time to work on this now. Given that, I included your suggested exception raising in the MappedModel instantiation if the mappingName is null.

Reply:
@spacether this is a potential NPE. You're protected in the constructor, but not in the property setter. This method should have a null check, something like this:

```
public int compareTo(MappedModel other) {
    if (null == getMappingName()) return 1;
    return getMappingName().compareTo(other.getMappingName());
```

Reply:
Why not protect the setter also?
Why would null be a valod value for a mappingName?

Reply:
@spacether it doesn't look like it should be an acceptable value in code, but this type is public with a getter/setter so it should cover all places where an NPE might be possible. It's not really common, and probably frowned upon to throw an exception in a property setter.

Reply:
I removed the exception raising.
This now returns int results for null and null to string comparisons.